src/cider/tests/nextgen/compiler/ContextTest.cpp

```diff
@@ -120,6 +125,89 @@ TEST_F(ContextTests, ContextBufferTest) {
   EXPECT_EQ(raw_buffer2[1], 2);
 }
 
+TEST_F(ContextTests, RegisterHashTableTest) {
+  CodegenContext codegen_ctx;
+  auto module = cider::jitlib::LLVMJITModule("test_register_hashtable", true);
+  cider::jitlib::JITFunctionPointer function =
+      cider::jitlib::JITFunctionBuilder()
+          .registerModule(module)
+          .setFuncName("register_hashtable")
+          .addReturn(cider::jitlib::JITTypeTag::VOID)
+          .addParameter(cider::jitlib::JITTypeTag::POINTER,
+                        "context",
+                        cider::jitlib::JITTypeTag::INT8)
+          .addProcedureBuilder([&codegen_ctx](cider::jitlib::JITFunctionPointer func) {
+            codegen_ctx.setJITFunction(func);
+            auto ret = codegen_ctx.registerHashTable("hash_table");
+            func->createReturn();
+          })
+          .build();
+  module.finish();
+
+  auto query_func = function->getFunctionPointer<void, int8_t*>();
+
+  auto input_builder = ArrowArrayBuilder();
+  auto&& [schema, array] =
+      input_builder.setRowNum(10)
+          .addColumn<int32_t>(
+              "a", CREATE_SUBSTRAIT_TYPE(I32), {1, 2, 3, 4, 5, 1, 2, 3, 4, 5})
+          .addColumn<int32_t>(
+              "b", CREATE_SUBSTRAIT_TYPE(I32), {0, 1, 2, 3, 4, 5, 6, 7, 8, 9})
+          .addColumn<int32_t>("c",
+                              CREATE_SUBSTRAIT_TYPE(I32),
+                              {100, 111, 222, 333, 444, 555, 666, 777, 888, 999})
+          .build();
+  StdMapDuplicateKeyWrapper<int, std::pair<Batch*, int>> dup_map;
+
+  Batch build_batch(*schema, *array);
+  cider_hashtable::
+      LinearProbeHashTable<int, std::pair<Batch*, int64_t>, murmurHash, Equal>
+          hm(16, 0);
+  auto batch = new Batch(build_batch);
+  for (int i = 0; i < 10; i++) {
+    int key = *(
+        (reinterpret_cast<int*>(const_cast<void*>(array->children[1]->buffers[1]))) + i);
+    hm.insert(key, std::make_pair(batch, i));
+    dup_map.insert(std::move(key), std::make_pair(batch, i));
+  }
+
+  // TODO(Xinyi) : sethashtable in velox hashjoinbuild
+  codegen_ctx.setHashTable(hm);
+  auto runtime_ctx = codegen_ctx.generateRuntimeCTX(allocator);
+
+  query_func((int8_t*)runtime_ctx.get());
+
+  EXPECT_EQ(runtime_ctx->getContextItemNum(), 1);
+  EXPECT_NE(runtime_ctx->getContextItem(0), nullptr);
+
+  auto hash_table = reinterpret_cast<
+      cider_hashtable::
+          LinearProbeHashTable<int, std::pair<Batch*, int>, murmurHash, Equal>*>(
+      runtime_ctx->getContextItem(0));
```

Comment:
why not provide a separate `getHashTablePtr` method instead? if you do register batch and register hashtable in codegen workflow, how to get correct index?

Reply:
This is mainly to ensure the consistency of the interface，like registerBatch and registerBuffer. and to get correct index, we have a acquireContextID() function above, This method is an auto-incrementer to make sure it doesn't repeat，and will save the only id in variable hashtable_descriptor_.

Reply:
I think the question is, you register this hashtable during translator and assign it an index, but during codegen, how do we know the correct index if there are multi hastables?

Reply:
Do you consider multi join nodes case? There will be multi `HashTableDescriptorPtr` and each one connects to a hash table. But how do we know which one we should use(the correct index) during codegen?

Reply:
Synced offline, seems multi join nodes are not considered now which means we assume that there is only one hash table for join.

Reply:
Yes, I think even in this case, just replace hashtable with vector<hashtable>,  register multiple times can also guarantee the uniqueness of its context_id by the time